Fix #3324: Adds SSL Certificate Viewer

Brandon-T opened the pull request:
## Security Review
- https://github.com/brave/security/issues/768

## Summary of Changes
- Depends on: https://github.com/brave/brave-core/pull/9565
- Adds Certificate Viewer to iOS with Brave-Core 1.37.x
- Tapping on the lock-icon in the URL bar displays the certificate information

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #3324

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
- Tap on the `lock` icon in the URL bar to show the leaf certificate information.


## Screenshots:
![image](https://user-images.githubusercontent.com/1530031/157102598-71f995ef-1df3-4d16-af47-b72b20fe528b.png)


## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

kylehickinson reviewed (changes requested):
Missed a few

thypon reviewed (changes requested):
> are there tests for this? seems important since it's security-related

With special care for `Client/Frontend/Browser/Certificate Viewer/BraveCertificateUtils.swift`

Brandon-T committed:
Certificate Viewer added.
Remade UI to look like design.

New Certificate Viewer code that uses Brave-Core's Chromium Parsing for Certificates.

Add unit tests

Add label to show certificate evaluation status.

Allow to show on the error interstitial page itself by creating a trust from the server chain.

Add DNT (Do Not Translate) strings.
